cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -187,14 +185,36 @@ public Replica replica(TopicPartition tp) {
   /**
    * Get number of replicas from the given topic in this broker.
    *
-   * @param topic Topic for which the replica count will be returned.
+   * @param topic Topic for which both the leader and follower replica count will be returned.
    * @return The number of replicas from the given topic in this broker.
    */
   public int numReplicasOfTopicInBroker(String topic) {
     Map<Integer, Replica> topicReplicas = _topicReplicas.get(topic);
     return topicReplicas == null ? 0 : topicReplicas.size();
   }
 
+  /**
+   * Get number of only leader replicas from the given topic in this broker.
+   *
+   * @param topicName Topic for which the replica count will be returned.
+   * @return The number of leader replicas from the given topic in this broker.
+   */
+  public int numLeaderReplicasOfTopicInBroker(String topicName) {
+    return numReplicasOfTopicInBroker(topicName, true);
+  }
+
+  /**
+   * Get number of either leader or follower replicas from the given topic in this broker
+   *
+   * @param topic Topic for which either the leader or the follower replica count will be returned.
+   * @param countLeaderReplica Count leader replicas if {@code true} and count follower replicas if
+   * {@code false}
+   * @return The number of replicas from the given topic in this broker.
+   */
+  private int numReplicasOfTopicInBroker(String topic, boolean countLeaderReplica) {
```

Comment:
Can we drop this function and move its content to `numLeaderReplicasOfTopicInBroker`? We are using it only as part of counting leaders, so filtering can be based on leadership.

Reply:
Yes, I dropped this method and keep:
```java
  public int numLeadersFor(String topicName) {
    return (int) replicasOfTopicInBroker(topicName).stream().filter(Replica::isLeader).count();
  }
```